eng/tools/npm-helpers.ps1

```diff
@@ -0,0 +1,162 @@
+$EngPath = Resolve-Path "${PSScriptRoot}/.."
+$Engcommon = Join-Path $EngPath "common"
+$EngCommonScriptsPath = Join-Path $Engcommon "scripts"
+. (Join-Path $EngCommonScriptsPath common.ps1)
+
+function GetNpmTagVersions($packageName)
+{
+  try
+  {
+    $existingVersion = Invoke-RestMethod -Method GET -Uri "http://registry.npmjs.com/${PackageName}"
+    $latest = ($existingVersion."dist-tags").latest
+    $next = ($existingVersion."dist-tags").next
+    Write-Host "Latest version: $latest"
+    Write-Host "Next version: $next"
+    if ($latest -eq $null) {
+      Write-Host "'latest' tag is not present in npm registry for package $packageName"
+    }
+    if ($next -eq $null) {
+      Write-Host "'Next' tag is not present in npm registry for package $packageName"
+    }
+
+    return New-Object PSObject -Property @{
+      latest = $latest
+      next = $next
+    }
+  }
+  catch
+  {
+    return $null
+  }
+}
+
+function GetNpmPackageVersions ($packageName)
+{
+  try
+  {
+    Write-Host "Checking versions present on npm for package $packageName"
+    $existingVersion = Invoke-RestMethod -Method GET -Uri "http://registry.npmjs.com/${packageName}"
+    return ($existingVersion.versions | Get-Member -MemberType NoteProperty).Name
+  }
+  catch
+  {
+    return $null
+  }
+}
+
+function FindRecentPackageVersion($packageName)
+{
+  $versions = (GetNpmPackageVersions -packageName $packageName) | ? {$_ -NotMatch "alpha|dev"}  
+  if ($versions -ne $null -and $versions.Count -gt 0)
+  {
+    $versions = [AzureEngSemanticVersion]::SortVersionStrings($versions)
+    $highestNpmVersion = $versions[0]
+    Write-Host "Recent version uploaded to NPM: $highestNpmVersion"
+    return $highestNpmVersion
+  }
+  
+  return $null
+}
+
+function GetNewNpmTags($packageName, $packageVersion)
+{
+  $newVersion = [AzureEngSemanticVersion]::ParseVersionString($packageVersion)
+  Write-Host "Package name: $packageName"
+  Write-Host "Package version: $packageVersion"
+  Write-Host "Find latest and next versions in npm registry for package"
+
+  <#
+  1. Get latest GA and latest preview from npm
+  2. Check new version to be published to find is it GA or preview
+  3. If new Version is GA:
+    a. If higher than current latest, or first GA then set LATEST tag
+  4. If new version is preview and higher than current higher version in npm:
+    a. Set LATEST if package has never GA released
+    b. Set NEXT tag
+  #>
+  $npmVersionInfo = GetNpmTagVersions -packageName $packageName
+  if ($npmVersionInfo -eq $null)
```

Comment:
Will we ever be in this state for new packages?

Reply:
Nope. I have removed this based on test cases run.